engine/AudioController.cpp

```diff
@@ -0,0 +1,95 @@
+#include "AudioController.h"
+#include <SDL/SDL_mixer.h>
+#include "Console.h"
+
+AudioController::AudioController()
+{
+	initSound();
+}
+
+bool AudioController::initSound()
+{
+	//Initialize SDL_mixer
+	if (Mix_OpenAudio(22050, MIX_DEFAULT_FORMAT, 2, 4096) == -1)
+	{
+		return false;
+	}
+	return true;
+}
+
+Mix_Music* AudioController::loadBackground(const char* file)
+{
+	//Load the music
+	Mix_Music* music = Mix_LoadMUS(file);
+
+	//If there was a problem loading the music
+	if (music == NULL)
+	{
+		Console::Println("Music not loaded");
+		return NULL;
+	}
+	Console::Println("Music loaded");
+	//If everything loaded fine
+	return music;
+}
+
+Mix_Chunk* AudioController::loadChunks(const char * file)
+{
+	//Load the sound effects
+	Mix_Chunk* chunk = Mix_LoadWAV(file);
+
+	//If there was a problem loading the sound effects
+	if (chunk == NULL)
+	{
+		return NULL;
+	}
+
+	//If everything loaded fine
+	return chunk;
+}
+
+void AudioController::cleanUp()
+{
+	//Free the music
+	//Mix_FreeMusic(music);
+}
+
+void AudioController::playSound(const char* file, int loop)
+{
+	if (loop < 0)
+	{
+		Mix_Music* music = loadBackground(file);
```

Comment:
Als dat zo is, controleer ook gelijk de rest van de file. er staan nog wat vergelijkbare dingen in